packages/core/src/fhir-deduplication/resources/medication.ts

```diff
@@ -76,7 +76,7 @@ export function groupSameMedications(medications: Medication[]): {
     } else if (snomedCode) {
       fillMaps(snomedMap, snomedCode, medication, refReplacementMap, false, removeOtherCodes);
     } else {
-      if (medication.id) danglingReferences.add(createMedicationRef(medication.id));
+      danglingReferences.add(createRef(medication));
```

Comment:
can we add a backupKey here for just the medication text? 
We dont want to throw out a med like this

```{
      "resource": {
        "resourceType": "Medication",
        ...
        "code": { "text": "lidocaine (PF) (XYLOCAINE) 392 mg in sodium chloride 0.9 % 50 mL IVPB" }
      }
    }

Reply:
or this

```    {
      "resource": {
        "resourceType": "Medication",
...
        "code": {
          "coding": [
            {
              "system": "http://www.nlm.nih.gov/research/umls/rxnorm",
              "display": "Microgestin FE 1/20 (28) 1 mg-20 mcg (21)/75 mg (7) tablet"
            }
          ]
        }
      }
    }

Reply:
so the backup key would be the code.text or the code.coding.display

Reply:
But then the issue is we wouldnt drop a med like this with that as the approach 

```    {
      "resource": {
        "resourceType": "Medication",
 ...
        ],
        "code": { "text": "No administered medications reported" }
      }
    }

Reply:
We could expand the blacklist to account for that?

Reply:
I think we can handle this as a separate PR, if that's good with you

Reply:
Sounds good I will take it asap